openmc/mesh.py

```diff
@@ -0,0 +1,560 @@
+
+        subelement = ET.SubElement(element, "lower_left")
+        subelement.text = ' '.join(map(str, self._lower_left))
+
+        if self._upper_right is not None:
+            subelement = ET.SubElement(element, "upper_right")
+            subelement.text = ' '.join(map(str, self._upper_right))
+
+        if self._width is not None:
+            subelement = ET.SubElement(element, "width")
+            subelement.text = ' '.join(map(str, self._width))
+
+        return element
+
+    @classmethod
+    def from_xml_element(cls, elem):
+        """Generate mesh from an XML element
+
+        Parameters
+        ----------
+        elem : xml.etree.ElementTree.Element
+            XML element
+
+        Returns
+        -------
+        openmc.Mesh
+            Mesh generated from XML element
+
+        """
+        mesh_id = int(get_text(elem, 'id'))
+        mesh = cls(mesh_id)
+
+        mesh_type = get_text(elem, 'type')
+        if mesh_type is not None:
+            mesh.type = mesh_type
+
+        dimension = get_text(elem, 'dimension')
+        if dimension is not None:
+            mesh.dimension = [int(x) for x in dimension.split()]
+
+        lower_left = get_text(elem, 'lower_left')
+        if lower_left is not None:
+            mesh.lower_left = [float(x) for x in lower_left.split()]
+
+        upper_right = get_text(elem, 'upper_right')
+        if upper_right is not None:
+            mesh.upper_right = [float(x) for x in upper_right.split()]
+
+        width = get_text(elem, 'width')
+        if width is not None:
+            mesh.width = [float(x) for x in width.split()]
+
+        return mesh
+
+    def build_cells(self, bc=['reflective'] * 6):
+        """Generates a lattice of universes with the same dimensionality
+        as the mesh object.  The individual cells/universes produced
+        will not have material definitions applied and so downstream code
+        will have to apply that information.
+
+        Parameters
+        ----------
+        bc : iterable of {'reflective', 'periodic', 'transmission', or 'vacuum'}
+            Boundary conditions for each of the four faces of a rectangle
+            (if aplying to a 2D mesh) or six faces of a parallelepiped
+            (if applying to a 3D mesh) provided in the following order:
+            [x min, x max, y min, y max, z min, z max].  2-D cells do not
+            contain the z min and z max entries.
+
+        Returns
+        -------
+        root_cell : openmc.Cell
+            The cell containing the lattice representing the mesh geometry;
+            this cell is a single parallelepiped with boundaries matching
+            the outermost mesh boundary with the boundary conditions from bc
+            applied.
+        cells : iterable of openmc.Cell
+            The list of cells within each lattice position mimicking the mesh
+            geometry.
+
+        """
+
+        cv.check_length('bc', bc, length_min=4, length_max=6)
```

Comment:
This seems to be the incorrect check. Since only 4 or 6 are valid values, cv.check_length needs to ensure it is 4 or 6, and 5 should fail.